dizoo/board_games/gomoku/envs/gomoku_env.py

```diff
@@ -0,0 +1,264 @@
+
+    def seed(self, seed: int, dynamic_seed: bool = True) -> None:
+        self._seed = seed
+        self._dynamic_seed = dynamic_seed
+        np.random.seed(self._seed)
+
+    def random_action(self):
+        action_list = self.legal_actions
+        return np.random.choice(action_list)
+
+    def render(self):
+        marker = "   "
+        for i in range(self.board_size):
+            if i <= 8:
+                marker = marker + self.board_markers[i] + "  "
+            else:
+                marker = marker + self.board_markers[i] + " "
+        print(marker)
+        for row in range(self.board_size):
+            if row <= 8:
+                print(str(1 + row) + ' ', end=" ")
+            else:
+                print(str(1 + row), end=" ")
+            for col in range(self.board_size):
+                ch = self.board[row][col]
+                if ch == 0:
+                    print(".", end="  ")
+                elif ch == 1:
+                    print("X", end="  ")
+                elif ch == 2:
+                    print("O", end="  ")
+            print()
+
+    def human_to_action(self):
+        """
+        For multiplayer games, ask the user for a legal action
+        and return the corresponding action number.
+        Returns:
+            An integer from the action space.
+        """
+        while True:
+            try:
+                row = int(
+                    input(
+                        f"Enter the row (1, 2, ...,{self.board_size}, from up to bottom) to play for the player {self.current_player}: "
+                    )
+                )
+                col = int(
+                    input(
+                        f"Enter the column (1, 2, ...,{self.board_size}, from left to right) to play for the player {self.current_player}: "
+                    )
+                )
+                choice = self.coord_to_action(row - 1, col - 1)
+                if (
+                        choice in self.legal_actions
+                        and 1 <= row
+                        and 1 <= col
+                        and row <= self.board_size
+                        and col <= self.board_size
+                ):
+                    break
+                else:
+                    print("Wrong input, try again")
+            except KeyboardInterrupt:
+                print("exit")
+                sys.exit(0)
+            except Exception as e:
+                print("Wrong input, try again")
+        return choice
+
+    def action_to_string(self, action_number):
+        """
+        Convert an action number to a string representing the action.
+
+        Args:
+            action_number: an integer from the action space.
+        Returns:
+            String representing the action.
+        """
+        row = action_number // self.board_size + 1
+        col = action_number % self.board_size + 1
+        return f"Play row {row}, column {col}"
+
+    def get_equi_data(self, play_data):
```

Comment:
rename it, and this function can be a common function of different chess games, you can implement it as a independent function and call it in different envs